ddtrace/appsec/processor.py

```diff
@@ -28,9 +30,20 @@
 log = get_logger(__name__)
 
 
-def _no_cookies(data):
-    # type: (Dict[str, str]) -> Dict[str, str]
-    return {key: value for key, value in data.items() if key.lower() not in ("cookie", "set-cookie")}
+def _transform_headers(data):
+    # type: (Dict[str, str]) -> Dict[str, Union[str, List[str]]]
+    normalized = {}  # type: Dict[str, Union[str, List[str]]]
+    for header, value in data.items():
+        header = header.lower()
+        if header in ("cookie", "set-cookie"):
+            continue
+        if header in normalized:  # if a header with the same lowercase name already exists, let's make it an array
+            if not isinstance(normalized[header], list):
+                normalized[header] = [normalized[header]]
+            normalized[header].append(value)
+            continue
+        normalized[header] = value
```

Comment:
```suggestion
        if header in normalized:  # if a header with the same lowercase name already exists, let's make it an array
            existing = normalized[header]
            if isinstance(existing, list):
                existing.append(value)
            else:
                normalized[header] = [existing, value]
        else:
            normalized[header] = value
```


Normally an `isinstance` check is enough to tell mypy what you mean, but it apparently doesn't carry through an item lookup.

This will also cut down on the dict lookups a little bit too.

Reply:
Thanks! It worked